app/tests/components_tests/test_models.py

```diff
@@ -1634,3 +1636,47 @@ def test_component_interface_value_manager():
 
     assert civ == civ1
     assert not created
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "kind,expected_queue",
+    (
+        (InterfaceKindChoices.STRING, False),
+        (InterfaceKindChoices.NEWICK, "acks-late-2xlarge"),
+    ),
+)
+def test_component_interface_custom_queue(kind, expected_queue, mocker):
+
+    ci = ComponentInterfaceFactory(
+        kind=kind,
+        store_in_database=False,
+    )
+    user = UserFactory()
+
+    # Need an existing CIVSet, use archive here since it is slightly easier setup
+    archive = ArchiveFactory()
+    archive.add_editor(user)
+    ai = ArchiveItemFactory.build(archive=None)
+
+    mock_task_signature = mocker.patch(
+        "grandchallenge.components.tasks.add_file_to_object.signature"
+    )
+    ai.validate_values_and_execute_linked_task(
+        values=[
+            CIVData(
+                interface_slug=ci.slug,
+                value=UserUploadFactory(creator=user),
+            )
+        ],
+        user=user,
+    )
+    assert mock_task_signature.called_once()
+
+    kwargs = mock_task_signature.call_args.kwargs
+    if not expected_queue:
+        # Ensure it's not passed since setting to None would
+        # have Celery use the Celery-scope default ("celerly")
+        assert "queue" not in kwargs
+    else:
+        assert kwargs["queue"] == expected_queue
```

Comment:
Reduces test logic:

```suggestion
    assert kwargs.get("queue") == expected_queue
```

The comment could then be moved to the parametrisation